From 6f69638c00fbe04e2ca7c606398b901b3dec930e Mon Sep 17 00:00:00 2001
From: Tiejun Chen <tiejun.chen@windriver.com>
Date: Wed, 5 Nov 2008 10:50:14 +0800
Subject: [PATCH] add flash support to emer_atcaf101

This creates the necessary of_device to the flash node on dts.

Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 arch/powerpc/sysdev/mv64x60_dev.c |   11 +++++++++++
 drivers/mtd/chips/jedec_probe.c   |   16 ++++++++++++++++
 2 files changed, 27 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/sysdev/mv64x60_dev.c b/arch/powerpc/sysdev/mv64x60_dev.c
index 32e0ad0..5d18c2a 100644
--- a/arch/powerpc/sysdev/mv64x60_dev.c
+++ b/arch/powerpc/sysdev/mv64x60_dev.c
@@ -494,6 +494,17 @@ static int __init mv64x60_device_setup(void)
 	for_each_compatible_node(np, NULL, "marvell,mv64360")
 		of_platform_bus_probe(np, of_mv64x60_devices, NULL);
 
+#ifdef CONFIG_PPC_PRPMC2800
+	/* Support Flash */
+	np = of_find_node_by_type (NULL, "rom");
+	if (np)
+		of_platform_device_create(np, "rom", NULL);
+
+	np = of_find_compatible_node(np, NULL, "jedec-flash");
+	if (np) 
+		of_platform_device_create(np, "jedec-flash", NULL);
+#endif
+
 	return 0;
 }
 arch_initcall(mv64x60_device_setup);
diff --git a/drivers/mtd/chips/jedec_probe.c b/drivers/mtd/chips/jedec_probe.c
index f84ab61..d351321 100644
--- a/drivers/mtd/chips/jedec_probe.c
+++ b/drivers/mtd/chips/jedec_probe.c
@@ -61,6 +61,7 @@
 #define AM29F002T	0x00B0
 #define AM29SL800DB	0x226B
 #define AM29SL800DT	0x22EA
+#define AM29LV008BB	0x0037
 
 /* Atmel */
 #define AT49BV512	0x0003
@@ -433,6 +434,21 @@ static const struct amd_flash_info jedec_table[] = {
 		}
 	}, {
 		.mfr_id		= MANUFACTURER_AMD,
+		.dev_id		= AM29LV008BB,
+		.name		= "AMD AM29LV008BB",
+		.devtypes	= CFI_DEVICETYPE_X16|CFI_DEVICETYPE_X8,
+		.uaddr		= MTD_UADDR_0x0555_0x02AA,
+		.dev_size	= SIZE_1MiB,
+		.cmd_set	= P_ID_AMD_STD,
+		.nr_regions	= 4,
+		.regions	= {
+			ERASEINFO(0x04000,1),
+			ERASEINFO(0x02000,2),
+			ERASEINFO(0x08000,1),
+			ERASEINFO(0x10000,15),
+		}
+	}, {
+		.mfr_id		= MANUFACTURER_AMD,
 		.dev_id		= AM29LV800BT,
 		.name		= "AMD AM29LV800BT",
 		.devtypes	= CFI_DEVICETYPE_X16|CFI_DEVICETYPE_X8,
-- 
1.5.5.1

